include/nbl/video/ILogicalDevice.h

```diff
@@ -252,7 +252,25 @@ class NBL_API ILogicalDevice : public core::IReferenceCounted, public IDeviceMem
                 return nullptr;
             if (!layout->wasCreatedBy(this))
                 return nullptr;
-            return createDescriptorSet_impl(pool, std::move(layout));
+
+            uint32_t descriptorCount[asset::EDT_COUNT] = { 0 };
+            for (const auto& binding : layout->getBindings())
+                descriptorCount[binding.type] += binding.count;
+
+            uint32_t descriptorStorageOffsets[asset::EDT_COUNT];
+            memset(descriptorStorageOffsets, 0xff, sizeof(descriptorStorageOffsets));
+
+            for (uint32_t type = 0; type < asset::EDT_COUNT; ++type)
+            {
+                if (descriptorCount[type] == 0)
+                    continue;
+
+                descriptorStorageOffsets[type] = pool->allocateDescriptors(static_cast<asset::E_DESCRIPTOR_TYPE>(type), descriptorCount[type]);
```

Comment:
allocate all `EDT_COUNT` at once using the method

Reply:
Resolved.